docs/zed/README.md

```diff
@@ -102,198 +292,357 @@ scans that run in the opposing order.
 
 Scans may also be range-limited but unordered.
 
-If data loaded into a pool lacks the pool key, that data is still
-imported but is not available to pool-key range scans.  Since it lacks
-the pool key, all data without a key is grouped together as a "null" key
-and cannot be efficiently range scanned.
+Any data loaded into a pool that lacks the pool key is presumed
+to have a null value with regard to range scans.  If large amounts
+of such "keyless data" are loaded into a pool, the ability to do
+range scans over such data is impaired.
+
+### 1.5 Time Travel
+
+Because commits are transactional and immutable, a query
+sees its entire data scan as a fixed "snapshot" with respect to the
+commit history.  In fact, Zed's [from operator](../zq/operators/from.md)
+allows a commit object to be specified with the `@` suffix to a
+pool reference, e.g.,
+```
+zed query 'from logs@1tRxi7zjT7oKxCBwwZ0rbaiLRxb | ...'
+```
+In this way, a query can time-travel through the commit history.  As long as the
+underlying data has not been deleted, arbitrarily old snapshots of the Zed
+lake can be easily queried.
+
+If a writer commits data after and while a reader is scanning, then the reader
+does not see the new data since it's scanning the snapshot that existed
+before these new writes occurred.
+
+Also, arbitrary metadata can be committed to the log as described below,
+e.g., to associate index objects or derived analytics to a specific
+journal commit point potentially across different data pools in
+a transactionally consistent fashion.
+
+While time travel through commit history provides one means to explore
+past snapshots of the commit history, another means is to use a timestamp.
+Because the entire history of branch updates is stored in a transaction journal
+and each entry contains a timestamp, branch references can be easily
+navigated by time.  For example, a list of branches of a pool's past
+can be created by scanning the branches log and stopping at the largest
```

Comment:
Yeah, it changes meaning (this would compute a snapshot in time of all the branch-to-commit-object mappings).  I will massage it a bit but maybe this is a design detail that we can take out later in the user-facing docs.